siege_utilities/testing/environment.py

```
@@ -230,8 +230,14 @@ def check_java_version() -> Tuple[Optional[str], bool]:
         >>>     print(f"Java {version} is compatible")
     """
     try:
+        # 10s timeout: a healthy `java -version` returns in milliseconds;
+        # a hang here cascades into diagnose_environment(),
+        # setup_spark_environment(), and quick_environment_setup(), all
+        # of which run during test setup. Bounded wait is required so a
+        # broken JDK install does not block the whole test invocation.
         result = subprocess.run(['java', '-version'],
```

Ruff (0.15.12)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)